pkg/controller/enterprisesearch/version_upgrade.go

```diff
@@ -201,7 +198,7 @@ func (r *VersionUpgrade) serviceURL() string {
 }
 
 // readOnlyModeRequest builds the HTTP request to toggle the read-only mode on Enterprise Search.
-func (r *VersionUpgrade) readOnlyModeRequest(enabled bool) (*http.Request, error) {
+func (r *VersionUpgrade) readOnlyModeRequest(ctx context.Context, enabled bool) (*http.Request, error) {
```

Comment:
This is the only substantial change I think. It also uncovered a couple places where we kind of take extra steps with the context passing both in entsrch and ES that could be simplified. In some place we build the request without a context, then we pass the context and request to a wrapper function that then adds the context and some headers to the request.

In a lot of cases we already have the context when we build the request, so we might as well build it with the context instead of adding it later. It makes it easier to tell if we are propagating contexts everywhere. Definitely a nitpicky thing but something we will probably want to do as we add context aware logging to existing controllers.